crates/eframe/src/web/mod.rs

```diff
@@ -136,10 +136,36 @@ fn get_canvas_element_by_id_or_die(canvas_id: &str) -> web_sys::HtmlCanvasElemen
 /// Returns the canvas in client coordinates.
 fn canvas_content_rect(canvas: &web_sys::HtmlCanvasElement) -> egui::Rect {
     let bounding_rect = canvas.get_bounding_client_rect();
+    // Need to fix Safari with vkeyboard open (flapping canvas position)
+    let rect_body = match web_sys::window()
+        .and_then(|w| w.document())
+        .and_then(|d| d.body())
+    {
+        None => egui::Rect::everything_above(0.),
```

Comment:
This seems like a bad fallback since it will produce infinities. When do you expect we will get here?